### PR TITLE
Change contribute to paddle to fit new branching model

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

**Comment by reyoung** [comment]
Related to #250

File: doc/build/contribute_to_paddle.md [hunk 1]
```diff
@@ -20,16 +20,30 @@ It's just that simple.
 
 ## Clone
```
Comment (backyes) [inline doc/build/contribute_to_paddle.md]: In original doc, 'Your code mush be fully documented by doxygen style.',  mush -> must?

File: doc/build/contribute_to_paddle.md [hunk 2]
```diff
@@ -20,16 +20,30 @@ It's just that simple.
...
+git clone --branch develop https://github.com/USERNAME/Paddle.git
+```
+If your repository doesn't contain **develop** branch, just create it by your own.
+
+```shell
+git clone https://github.com/USERNAME/Paddle.git Paddle
+cd Paddle
+git checkout -b develop  # create develop branch.
+git remote add upstream https://github.com/baidu/Paddle.git  # add upstream to baidu/Paddle
+git pull upstream develop  # update to upstream
```
Comment (backyes) [inline doc/build/contribute_to_paddle.md]: 会不会fetch + merge 比 pull 靠谱。

File: doc/build/contribute_to_paddle.md [hunk 3]
```diff
@@ -63,7 +77,7 @@ git remote -v
 Update your fork with the latest upstream changes:
```
Comment (backyes) [inline doc/build/contribute_to_paddle.md]: ## Keeping Fork Up to Date
 
 Before pull your request, you should sync your code from the latest PaddlePaddle.
 To do this, you'll need to add a remote at first:
 
 ```shell
 # see the current configured remote repository
 git remote -v
 # add upstream repository
 git remote add upstream https://github.com/baidu/Paddle.git
 # verify the new upstream
 git remote -v
 ```
 
 # add upstream repository
 git remote add upstream https://github.com/baidu/Paddle.git

前面已经描述过了，可以删掉这行。

File: doc/build/contribute_to_paddle.md [hunk 4]
```diff
@@ -93,10 +107,10 @@ of conflict, you need to do the update manually. You need to do the following on
 your local repository:
```
Comment (backyes) [inline doc/build/contribute_to_paddle.md]: * pull request之后 加一个 follow comment的会比较好。 要反复重复comit + fetch + merge + push 的操作。 

* 考虑加一个 refer： https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes ？

File: doc/build/contribute_to_paddle.md (hunk 2 above)
Comment (reyoung) [inline doc/build/contribute_to_paddle.md]: 不会。

`git fetch + git merge == git pull`

**Pull request merged**